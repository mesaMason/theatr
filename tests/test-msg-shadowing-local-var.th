dolphin(int weight, int age):
    print("inside dolphin.")
    print("dolphins local sum var =")
    int sum = weight + age
    print(sum)

    receive:
        swim():
            print("inside swim().")
            print("dolphins local sum var =")
            print(sum)
        eat():
            int sum = 0
            print("inside eat().")
            print("eat()s local sum var =")
            print(sum)

    drop:
        print("inside drop().")
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    arwgadg() | d
    eat() | d
    swim() | d
    die() | d
    return 0

