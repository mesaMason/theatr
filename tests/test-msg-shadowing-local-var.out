inside dolphin.
dolphin's local sum var =
13
inside drop().
inside eat().
eat()'s local sum var =
0
inside swim().
dolphin's local sum var =
13
